File: elixir/captains-log/lib/captains_log.ex
defmodule CaptainsLog do
  @planetary_classes ["D", "H", "J", "K", "L", "M", "N", "R", "T", "Y"]

  def random_planet_class() do
    Enum.random(@planetary_classes)
  end

  def random_ship_registry_number() do
    "NCC-#{Enum.random(1000..10000)}"
  end

  def random_stardate() do
    41000 + :rand.uniform() * 1000
  end

  def format_stardate(stardate) do
    :io_lib.format("~.1f", [stardate])
    |> :erlang.iolist_to_binary()
  end
end

File: java/captains-log/src/main/java/CaptainsLog.java
import java.util.Random;

class CaptainsLog {

    private static final char[] PLANET_CLASSES = new char[]{'D', 'H', 'J', 'K', 'L', 'M', 'N', 'R', 'T', 'Y'};

    private Random random;

    CaptainsLog(Random random) {
        this.random = random;
    }

    char randomPlanetClass() {
        return PLANET_CLASSES[this.random.nextInt(PLANET_CLASSES.length)];
    }

    String randomShipRegistryNumber() {
        final int shipRangeStart = 1000;
        final int shipRangeEnd = 10000;

        return "NCC-" + this.random.nextInt(shipRangeStart, shipRangeEnd);
    }

    double randomStardate() {
        final int stardateRangeStart = 41000;
        final int stardateRangeEnd = 42000;

        return this.random.nextDouble(stardateRangeStart, stardateRangeEnd);
    }
}
